libsolidity/ast/ASTAnnotations.h

```diff
@@ -312,7 +312,12 @@ struct MemberAccessAnnotation: ExpressionAnnotation
 	util::SetOnce<VirtualLookup> requiredLookup;
 };
 
-struct BinaryOperationAnnotation: ExpressionAnnotation
+struct OperationAnnotation: ExpressionAnnotation
+{
+	FunctionDefinition const* userDefinedFunction = nullptr;
```

Comment:
It might work to use `referencedDeclaration`, but we should have a clear distinction that states if this is a function call or not. I'm ok with renaming, but maybe both could be done when we change this for the literal suffixes?

Reply:
ok, I guess I can still change it with literal suffixes.

Still, here I'd at least give it a better name just in case it does happen to stay as is. So how about `operatorDefinition`? `boundFunction`? `customImplementation`?